lib/url.c

```diff
@@ -3992,7 +3992,10 @@ CURLcode Curl_setup_conn(struct connectdata *conn,
   }
   else {
     Curl_pgrsTime(data, TIMER_CONNECT);    /* we're connected already */
-    Curl_pgrsTime(data, TIMER_APPCONNECT); /* we're connected already */
+    if(conn->ssl[FIRSTSOCKET].use ||
+       ((conn->handler->protocol & PROTO_FAMILY_SSH) &&
+        conn->proto.sshc.authed))
```

Comment:
I don't think "conn->proto.sshc.authed" is necessary here. If the SSH protocol family is used, there *will* be an appconnect phase.

Reply:
> I don't think "conn->proto.sshc.authed" is necessary here. If the SSH protocol family is used, there _will_ be an appconnect phase.

In the code appconnect timer is only set when authed is complete. So there will be a phase but could the connection be reused without authed being complete?

Reply:
No, it shouldn't be possible. When the connection is put into the connection pool, it is either authenticated or it failed (and thus wasn't put into the pool to begin with).